tests/test_artifactory.py

```diff
@@ -125,3 +127,34 @@ def test_errors(tmpdir, backend):
         remote_file,
         suppress_backend_errors=True,
     ) == []
+
+
+@pytest.mark.parametrize(
+    'backend',
+    ['artifactory'],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    'file, version, extensions, expected',
+    [
+        ('/file.tar.gz', '1.0.0', None, 'file.tar/1.0.0/file.tar-1.0.0.gz'),
```

Comment:
I would add one additional test here to make sure we also cover that case:

```suggestion
        ('/file.tar.gz', '1.0.0', None, 'file.tar/1.0.0/file.tar-1.0.0.gz'),
        ('/file.tar.gz', '1.0.0', [], 'file.tar/1.0.0/file.tar-1.0.0.gz'),
```